pkg/container-utils/docker/docker.go

```diff
@@ -123,30 +100,68 @@ func (c *DockerClient) GetContainers() ([]*runtimeclient.ContainerData, error) {
 }
 
 func (c *DockerClient) GetContainer(containerID string) (*runtimeclient.ContainerData, error) {
-	filter := dockerfilters.NewArgs()
-	filter.Add("id", containerID)
+	containerID, err := runtimeclient.ParseContainerID(Name, containerID)
+	if err != nil {
+		return nil, err
+	}
 
-	containers, err := listContainers(c, &filter)
+	containerJSON, err := c.client.ContainerInspect(context.Background(), containerID)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(containers) == 0 {
-		return nil, fmt.Errorf("container %q not found", containerID)
+	if containerJSON.State == nil {
+		return nil, errors.New("container state is nil")
 	}
-	if len(containers) > 1 {
-		log.Warnf("DockerClient: multiple containers (%d) with ID %q. Taking the first one: %+v",
-			len(containers), containerID, containers)
+	if containerJSON.State.Pid == 0 {
+		return nil, errors.New("got zero pid")
+	}
+	if containerJSON.HostConfig == nil {
+		return nil, errors.New("container host config is nil")
+	}
+	containerData := runtimeclient.ContainerData{
+		ID:      containerJSON.ID,
+		Name:    strings.TrimPrefix(containerJSON.Name, "/"),
+		State:   containerStatusStateToRuntimeClientState(containerJSON.State.Status),
+		Runtime: Name,
 	}
 
-	return DockerContainerToContainerData(&containers[0]), nil
+	containerExtraInfo := runtimeclient.ContainerExtraInfo{
+		Pid:         containerJSON.State.Pid,
+		CgroupsPath: string(containerJSON.HostConfig.Cgroup),
+	}
+	containerExtraInfo.Mounts = []runtimeclient.ContainerMountData{}
+	for _, containerMount := range containerJSON.Mounts {
+		containerExtraInfo.Mounts = append(containerExtraInfo.Mounts, runtimeclient.ContainerMountData{
+			Destination: containerMount.Destination,
+			Source:      containerMount.Source,
+		})
+	}
+
+	// Check if cgroups path is empty -- if so, it will need to be read from /proc/<pid>/cgroup.
```

Comment:
```suggestion 
        // Try to get cgroups information from /proc/<pid>/cgroup as a fallback.
	// However, don't fail if such a file is not available, as it would prevent the
	// whole feature to work on systems without this file.
```

WDYT?